src/kudu/consensus/leader_election.h

```diff
@@ -41,7 +41,30 @@ namespace consensus {
 // The vote a peer has given.
 enum ElectionVote {
   VOTE_DENIED = 0,
-  VOTE_GRANTED = 1,
+  VOTE_GRANTED = 1
+};
+
+// Details of the vote received from a peer.
+struct VoteInfo {
+  ElectionVote vote;
+
+  // Previous voting history of this voter.
+  std::vector<PreviousVotePB> previous_vote_history;
+};
+
+// Internal structure to denote the optimizer's computation of the
+// next potential leader.
+struct PotentialNextLeaderResponse {
+  enum Status {
+    POTENTIAL_NEXT_LEADER_DETECTED = 0,
```

Comment:
it would be nice to explain the states.

Reply:
I thought they were self explanatory but I am working on a doc. Will publish it soon and add it here as well.